ext/FHistHDF5Ext.jl

```diff
@@ -0,0 +1,85 @@
+module FHistHDF5Ext
+
+using FHist
+using HDF5, StatsBase
+using Base.Threads: SpinLock
+
+import FHist: h5readhist, h5dumphist
+
+
+"""
+    function h5dumphist(filename::AbstractString, path::AbstractString, h::Hist1D)
+
+Writes a [`Hist1D`](@ref) instance to an HDF5 file. The histogram can be retrieved
+using [`h5readhist`](@ref).
+
+# Examples
+h = Hist1D(rand(1000), -3:0.3:3)
+h5dumphist("foo.h5", "some/path/to/myhist", h)
+"""
+function h5dumphist(filename::AbstractString, path::AbstractString, h::Union{Hist1D, Hist2D, Hist3D})
+    h5open(filename, "cw") do f
+        g = create_group(f, path)
+        write(g, "weights", h.hist.weights)
+        write(g, "sumw2", h.sumw2)
+        for (dim, edges) in enumerate(h.hist.edges)
+            write(g, "edges_$dim", collect(edges))
+        end
+        attributes(g)["isdensity"] = h.hist.isdensity
+        attributes(g)["closed"] = string(h.hist.closed)
+        attributes(g)["overflow"] = string(h.overflow)
+        attributes(g)["nentries"] = h.nentries.x
+        attributes(g)["_producer"] = "FHist.jl"
```

Comment:
do you want to version this? like what promise are we making here? Because I think it's likely we will break the content in the future.

For example, we don't actually manipuate `isdensity` and `closed` in FHist.jl, you may not want to store them

Reply:
Yes, I thought about versioning, that's actually something I wanted to discuss ;) We can add a simple version number for now...

Even if we don't manipulate `isdensity` or `closed`, those are still attributes we want to store, don't we?

Reply:
Not really because I don't plan to use them. So I was thinking in the read() function, you just put in the default values for these two attributes.

In the next version we can just remove the usage of StatsBase Histogram